client/src/app/shared/components/chip/chip.component.ts

```diff
@@ -0,0 +1,78 @@
+import { Component, ElementRef, Input, OnInit } from '@angular/core';
+
+import { HtmlColor } from 'app/core/definitions/key-types';
+
+@Component({
+    selector: 'os-chip',
+    templateUrl: './chip.component.html',
+    styleUrls: ['./chip.component.scss']
+})
+export class ChipComponent {
+    @Input()
+    public set color(color: HtmlColor) {
+        if (!(color.length === 3 || color.length === 6 || color.length === 4 || color.length === 7)) {
+            throw new Error(`Undefined color: ${color}`);
+        }
+        this._color = color;
+        this.recalcForegroundColor();
+    }
+
+    public get color(): HtmlColor {
+        return this._color;
+    }
+
+    private get red(): string {
```

Comment:
Can you put this into a "color-service" dealing with parsing/normalizing/generating color values?